File: 0322-coin-change/0322-coin-change.go
import "math"

func coinChange(coins []int, amount int) int {
    
    dp := make([]int, amount+1)
    for i := 0; i <= amount ;i++ {
        dp[i] = math.MaxInt32;
    }
    dp[0] = 0

    for i := 1  ; i <= amount ; i++ {
        //코인의 갯수만큼
        for _,coin := range coins {
            if i - coin >= 0 { //코인을 만들 수 있으면,
                dp[i] = min(dp[i], dp[i-coin]+1)
            }
        }
    }

    if dp[amount] == math.MaxInt32 {
        return -1
    }
    return dp[amount]
    
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: 0322-coin-change/0322-coin-change.js
var coinChange = function(coins, amount) {
    let dp = new Array(amount + 1).fill(Infinity);
    dp[0] = 0; // 금액 0은 동전 0개로 만들 수 있음.

    for (let i = 1; i <= amount; i++) {
        for (let coin of coins) {
            if (i - coin >= 0) { // 현재 동전으로 만들 수 있는 경우
                dp[i] = Math.min(dp[i], dp[i - coin] + 1);
            }
        }
    }
    return dp[amount] === Infinity ? -1 : dp[amount];
};